src/consensus/pbft_manager.cpp

```diff
@@ -675,12 +817,18 @@ void PbftManager::identifyBlock_() {
       LOG(log_dg_) << "Identify leader block " << leader_block.first << " at round " << round;
 
       auto place_votes = placeVote_(leader_block.first, soft_vote_type, round, step_);
+
+      checkSoftVotedValueChange_(leader_block.first);
+
       if (place_votes) {
         LOG(log_nf_) << "Soft votes " << place_votes << " voting block " << leader_block.first << " at round " << round;
       }
     }
   } else if (round >= 2 && previous_round_next_voted_value_ != NULL_BLOCK_HASH) {
     auto place_votes = placeVote_(previous_round_next_voted_value_, soft_vote_type, round, step_);
+
+    checkSoftVotedValueChange_(previous_round_next_voted_value_);
```

Comment:
I don't understand here. Here will always propose previous round next voted value, never change.
Maybe you want to terminate the soft voted value if after max waiting time?